rasa/nlu/featurizers/count_vectors_featurizer.py

```diff
@@ -220,51 +307,109 @@ def train(
             # create spacy lemma_ for OOV_words
             self.OOV_words = [t.lemma_ for w in self.OOV_words for t in spacy_nlp(w)]
 
-        self.vectorizer = CountVectorizer(
-            token_pattern=self.token_pattern,
-            strip_accents=self.strip_accents,
-            lowercase=self.lowercase,
-            stop_words=self.stop_words,
-            ngram_range=(self.min_ngram, self.max_ngram),
-            max_df=self.max_df,
-            min_df=self.min_df,
-            max_features=self.max_features,
-            analyzer=self.analyzer,
+        self.vectorizer = self.create_vectorizers(
+            self.token_pattern,
+            self.strip_accents,
+            self.lowercase,
+            self.stop_words,
+            (self.min_ngram, self.max_ngram),
+            self.max_df,
+            self.min_df,
+            self.max_features,
+            self.analyzer,
+            shared=self.use_shared_vocab,
         )
 
-        lem_exs = [
-            self._get_message_text(example) for example in training_data.intent_examples
-        ]
+        cleaned_attribute_texts = {}
+
+        for attribute in MESSAGE_ATTRIBUTES:
+            attribute_texts = [
+                self._get_message_text_by_attribute(example, attribute)
+                if example.get(attribute) is not None
+                else ""
+                for example in training_data.intent_examples
+            ]
+            self._check_OOV_present(attribute_texts)
+            cleaned_attribute_texts[attribute] = attribute_texts
+
+        combined_cleaned_texts = []
+        if self.use_shared_vocab:
+            for attribute in MESSAGE_ATTRIBUTES:
+                combined_cleaned_texts += cleaned_attribute_texts[attribute]
+
+        featurized_attributes = {}
+        # noinspection PyPep8Naming
+
+        for index, attribute in enumerate(MESSAGE_ATTRIBUTES):
+            try:
+                if self.use_shared_vocab:
+                    if index == 0:
+                        # Only train a model for first attribute
+                        self.vectorizer[attribute].fit(combined_cleaned_texts)
+                    else:
+                        # Copy the first model for this attribute
+                        self.vectorizer[attribute] = self.vectorizer[
+                            MESSAGE_ATTRIBUTES[0]
+                        ]
+                else:
+                    self.vectorizer[attribute].fit(cleaned_attribute_texts[attribute])
+            except ValueError:
+                logger.warning(
+                    "Unable to train CountVectorizer for message attribute {0}. "
```

Comment:
minor
```suggestion
                    "Unable to train CountVectorizer for message attribute {}. "
```

Reply:
I have this format in a lot of places. Do you recommend changing it everywhere? 😄

Reply:
You could 😄 There is not need to add the `0` if you don't want to place one variable multiple times in the string. We normally don't add it.